src/array/merge.ts

```diff
@@ -31,9 +31,10 @@ export function merge<T>(
   if (!toKey) {
     return [...prev]
   }
-  const keys = array.map(toKey)
+  const keys = new Map(array.map(v => [toKey(v), v]))
```

Comment:
Should we call `keys.set` in a loop instead of allocating all of these arrays? Less GC work then, though I'm not sure if that offsets the cost of resizing the Map for large arrays. 🤔

Reply:
I've pushed a new commit with proposed change. You can compare the benchmarks. I think values are pretty unstable and difference is not big. But current version indeed avoids creating an extra array. It's hard to say which one is better, but the latest version feels right.

Reply:
Agreed. IDK if GC work is properly measured in benchmarks or not either.